src/js/components/DataTable/Body.js

```diff
@@ -198,16 +242,22 @@ const Body = forwardRef(
     ref,
   ) => {
     const { theme, passThemeFlag } = useThemeValue();
+    const [focused, setFocused] = React.useState();
+    const [lastFocused, setLastFocused] = React.useState();
     const [active, setActive] = React.useState();
-    const [lastActive, setLastActive] = React.useState();
     const [scroll, setScroll] = React.useState();
     const containerRef = useForwardedRef(ref);
 
-    // Determine if using a keyboard to cover focus behavior
-    const usingKeyboard = useKeyboard();
-
-    const onFocusActive =
-      active ?? lastActive ?? (usingKeyboard && onClickRow ? 0 : undefined);
+    // Store refs for each row to enable direct focus management
+    const rowRefs = React.useRef(new Map());
```

Comment:
It feels a little heavy to create refs for all the rows, wondering if we are able to do something similar to roving tab index in Calendar where we just have a ref for the item that can receive focus

Reply:
I believe [each row already has a ref](https://github.com/grommet/grommet/blob/master/src/js/components/DataTable/Body.js#L43) -- we're just taking that and storing in a map to know what to focus. I can look into another approach where the focusableRow ref is just a single row and we only apply it in the case where tabIndex=0 if you'd prefer though.